src/Uno.UWP/UI/Input/GestureRecognizer.cs

```diff
@@ -0,0 +1,250 @@
+using Microsoft.Extensions.Logging;
+using Uno.Extensions;
+using Uno.Logging;
+
+namespace Windows.UI.Input
+{
+	public  partial class GestureRecognizer 
+	{
+		private readonly ILogger _log;
+		private IDictionary<uint, List<PointerPoint>> _activePointers = new Dictionary<uint, List<PointerPoint>>();
+
+		public GestureSettings GestureSettings { get; set; }
+
+		public bool IsActive => _activePointers.Count > 0;
+
+		public GestureRecognizer()
+		{
+			_log = this.Log();
+		}
+
+		public void ProcessDownEvent(PointerPoint value)
+		{
+			if (TryRecognizeMultiTap(value))
+			{
+				// This is how UWP behaves, it will fire the double tap
+				// on Down for a double tap instead of the Up.
+
+				return;
+			}
+
+			_activePointers[value.PointerId] = new List<PointerPoint>(16) { value }; ;
+		}
+
+		public void ProcessMoveEvents(IList<PointerPoint> value)
+		{
+			foreach (var point in value)
+			{
+				if (_activePointers.TryGetValue(point.PointerId, out var points))
+				{
+					points.Add(point);
+				}
+				else if (_log.IsEnabled(LogLevel.Information))
+				{
+					// info: We might get some PointerMove for mouse even if not pressed!
+					_log.Info("Received a 'Move' for a pointer which was not considered as down. Ignoring event.");
+				}
+			}
+		}
+		public void ProcessUpEvent(PointerPoint value)
+		{
+#if NET461 || __WASM__
+			if (_activePointers.TryGetValue(value.PointerId, out var points))
+			{
+				_activePointers.Remove(value.PointerId);
+#else
+			if (_activePointers.Remove(value.PointerId, out var points))
+			{
+#endif
+
+				// Note: At this point we MAY be IsActive == false, which is the expected behavior (same as UWP)
+				//		 even if we will fire some events now.
+
+				Recognize(points, pointerUp: value);
+			}
+			else if (_log.IsEnabled(LogLevel.Error))
+			{
+				_log.Error("Received a 'Up' for a pointer which was not considered as down. Ignoring event.");
+			}
+		}
+
+		public void CompleteGesture()
+		{
+			if (!IsActive)
+			{
+				return;
+			}
+
+			// Capture the list in order to avoid alteration while enumerating
+			var current = _activePointers;
+			_activePointers = new Dictionary<uint, List<PointerPoint>>();
+
+			// Note: At this point we are IsActive == false, which is the expected behavior (same as UWP)
+			//		 even if we will fire some events now.
+
+			// Complete all pointers
+			foreach (var points in current.Values)
+			{
+				Recognize(points, pointerUp: null);
+			}
+		}
+
+		private void Recognize(List<PointerPoint> points, PointerPoint pointerUp)
+		{
+			if (points.Count == 0)
+			{
+				return;
+			}
+
+			var recognized = TryRecognizeTap(points, pointerUp);
+
+			if (!recognized && _log.IsEnabled(LogLevel.Information))
+			{
+				_log.Info("No gesture recognized");
+			}
+		}
+
+		#region Tap (includes DoubleTap)
+		internal const ulong MultiTapMaxDelayTicks = TimeSpan.TicksPerMillisecond * 1000;
+		internal const int TapMaxXDelta = 15;
+		internal const int TapMaxYDelta = 15;
```

Comment:
I think this value should be calculated using the physical DPI of the device.

Reply:
Those values are kind of random values based on my tests. If at some point we feel the need to fine-tune them for some specific platform / DPI it will be easy as I extract them, but currently I didn't notice any issue and I don't see how I could improve them.

I would however argue about the need to fine tune them based on the DPI as we are only working with virtual pixels at this point, so the DPI was already taken in consideration.

I'll close this comment as currently there is no actionable item.